config/crd/bases/capabilities.3scale.net_backends.yaml

```diff
@@ -125,6 +125,9 @@ spec:
                 description: SystemName identifies uniquely the backend within the
                   account provider Default value will be sanitized Name
                 type: string
+                x-kubernetes-validations:
```

Comment:
Hi @carlkyrillos , this approach to make field immutable looks ok, but I'd like to ask if you compared it with K8S "native" way , as following 
```
                systemName:
                  type: string
                  x-kubernetes-immutable: true
```
I asked ChatGPT to compare both approaches, and this is table below

### **Comparison of Both Approaches**

| **Feature**                  | **`x-kubernetes-immutable: true` (in CRD)**                     | **`+kubebuilder:validation:XValidation` (in Go)**              |
|------------------------------|------------------------------------------------------------------|-----------------------------------------------------------------|
| **Where is the validation enforced?** | Kubernetes API server (global enforcement)                   | Operator (custom code in controller-runtime)                     |
| **Scope of enforcement**      | Validates immutability across all Kubernetes clients and tools (kubectl, API, etc.) | Validates only when the operator is active and running          |
| **Ease of implementation**    | Simple, no code needed outside the CRD definition               | Requires adding validation logic to the Go code (operator)      |
| **Customization**             | No customization beyond immutability                           | Can customize the validation rule and error message             |
| **Error handling**            | Automatically handled by the Kubernetes API server             | You need to handle errors manually in the operator code         |
| **Flexibility**               | Only supports basic immutability                               | Supports complex validation logic (e.g., conditional immutability) |
| **Operational dependencies**  | No dependency on the operator or controller                    | Depends on the operator running to enforce validation          |

Reply:
I checked solution. It's working.

Reply:
Regarding second way I noted before. Checked again, and meanwhile it's not working for me, option looks like not accepted. I also more checked google, Not ChatGpt , and I see your approach. So it's good. Thank you